This addresses build failures like:
Reason: Task ':maven-extension:jApiCmp' uses this output of task ':maven-extension:shadowJar' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.

diff --git a/maven-extension/build.gradle.kts b/maven-extension/build.gradle.kts
index 15755f0..0fec1b4 100644
--- a/maven-extension/build.gradle.kts
+++ b/maven-extension/build.gradle.kts
@@ -63,3 +63,6 @@ tasks {
 }
 
 tasks.getByName("test").dependsOn("shadowJar")
+afterEvaluate {
+  tasks.getByName("jApiCmp").dependsOn("shadowJar")
+}
